pkgs/stdenv/generic/setup.sh

```diff
@@ -1,7 +1,7 @@
 set -eu
 set -o pipefail
 
-if [ -n "${BASH_VERSINFO-}" ] && [ "${BASH_VERSINFO-}" -lt 4 ]; then
+if [[ -n "${BASH_VERSINFO-}" && "${BASH_VERSINFO-}" -lt 4 ]]; then
```

Comment:
Does it count as SC 2166 if the code you start with is already what shellcheck considers correct?

Reply:
You're right here. This was part of something else. Happy to revert.

Reply:
Not sure about something else, there are two more changes like this in this PR… (and one change that got converted from a true SC 2166 to a logic simplification)

Reply:
Yes, I would like to bring in the logic simplification part, I don't mind renaming the commit if you want.
I don't mind removing the `[[` -> `[`.
Let me know what you think should be removed and I can have a look.

Reply:
I would expect logic simplification to be in a commit mentioning logic simplification, and `[` to `[[` conversion can happen, but then it should be called something about conditional style. I just don't think like having a commit called SC 2166 doing two different things, of which neither is actually a fix for SC 2166, is a good idea.

(Having a single commit with a long commit message is also fine — I just want to avoid an actively misleading commit title)

Reply:
Let me amend the commit message then.

Reply:
I've tried to be more descriptive, let me know if you still think it needs work.

Reply:
The original commit now looks a bit pointless… The rest is good, in my opinion

Reply:
yeah I missed it. Let me just squash those.